tests/block_header_test.go

```diff
@@ -155,3 +184,89 @@ func testHeaders_BaseFeeEvolutionFollowsPricingRules(t *testing.T, headers []*ty
 		)
 	}
 }
+
+func testHeaders_TransactionRootMatchesBlockTxsHash(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		block, err := client.BlockByNumber(context.Background(), header.Number)
+		require.NoError(err, "failed to get block receipts")
+
+		txsHash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil))
+		require.Equal(header.TxHash, txsHash, "transaction root hash mismatch")
+	}
+}
+
+func testHeaders_ReceiptRootMatchesBlockReceipts(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(header.Number.Uint64())))
+		require.NoError(err, "failed to get block receipts")
+
+		receiptsHash := types.DeriveSha(types.Receipts(receipts), trie.NewStackTrie(nil))
+		require.Equal(header.ReceiptHash, receiptsHash, "receipt root hash mismatch")
+	}
+}
+
+func testHeaders_LogsBloomMatchesLogsInReceipts(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(header.Number.Uint64())))
+		require.NoError(err, "failed to get block receipts")
+
+		logsBloom := types.CreateBloom(receipts)
+		require.Equal(header.Bloom, logsBloom, "logs bloom mismatch")
+	}
+}
+
+func testHeaders_CoinbaseIsZeroForAllBlocks(t *testing.T, headers []*types.Header) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		require.Empty(header.Coinbase, "coinbase is not zero")
```

Comment:
nit: Is `Empty` checking for a zero value here? It would be cleaner to do a `require.Equal` here.

Reply:
yes, it does check for zero value. There is also a `.Zero` method, I'll use that one.